src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
+        env.trust(Account("dan")["USD"](20), "bob");
+        env.trust(Account("alice")["USD"](20), "carol");
+        env.trust(Account("carol")["USD"](20), "dan");
+        env(pay("alice", "bob", Account("bob")["USD"](55)),
+            paths(Account("alice")["USD"]));
+        env.require(balance("bob", Account("alice")["USD"](40)));
+        env.require(balance("bob", Account("dan")["USD"](15)));
+    }
 
-        // Payment with path.
-        payWithPath(alice, bob, "USD", "24", ledger);
+    // alice -120 USD-> amazon -25 USD-> bob
+    // alice -25 USD-> carol -75 USD -> dan -100 USD-> bob
+    void
+    test_issues_path_negative_ripple_client_issue_23_larger()
+    {
+        testcase("path negative: ripple-client issue #23: larger");
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob", "carol", "dan", "edward");
+        env.trust(Account("alice")["USD"](120), "edward");
+        env.trust(Account("edward")["USD"](25), "bob");
+        env.trust(Account("dan")["USD"](100), "bob");
+        env.trust(Account("alice")["USD"](25), "carol");
+        env.trust(Account("carol")["USD"](75), "dan");
+        env(pay("alice", "bob", Account("bob")["USD"](50)),
+            paths(Account("alice")["USD"]));
+        env.require(balance("alice", Account("edward")["USD"](-25)));
+        env.require(balance("alice", Account("carol")["USD"](-25)));
+        env.require(balance("bob", Account("edward")["USD"](25)));
+        env.require(balance("bob", Account("dan")["USD"](25)));
+        env.require(balance("carol", Account("alice")["USD"](25)));
+        env.require(balance("carol", Account("dan")["USD"](-25)));
+        env.require(balance("dan", Account("carol")["USD"](25)));
+        env.require(balance("dan", Account("bob")["USD"](-25)));
+    }
 
-        // Verify balances
-        verifyBalance(ledger, alice, Amount(46, "USD", mtgox));
-        verifyBalance(ledger, mtgox, Amount(-46, "USD", alice));
-        verifyBalance(ledger, mtgox, Amount(-24, "USD", bob));
-        verifyBalance(ledger, bob, Amount(24, "USD", mtgox));
+    // carol holds mtgoxAUD, sells mtgoxAUD for XRP
+    // bob will hold mtgoxAUD
+    // alice pays bob mtgoxAUD using XRP
+    void
+    test_via_offers_via_gateway()
+    {
+        testcase("via gateway");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const AUD = gw["AUD"];
+        env.fund(XRP(10000), "alice", "bob", "carol", gw);
+        env(rate(gw, 1.1));
+        env.trust(AUD(100), "bob");
+        env.trust(AUD(100), "carol");
+        env(pay(gw, "carol", AUD(50)));
+        env(offer("carol", XRP(50), AUD(50)));
+        env(pay("alice", "bob", AUD(10)), sendmax(XRP(100)), paths(XRP));
+        env.require(balance("bob", AUD(10)));
+        env.require(balance("carol", AUD(39)));
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"](0)}, Account("bob")["USD"](25));
```

Comment:
I'm not sure why this `findPath` belongs with this test. What is this meant to be testing?